pytorch_lightning/strategies/ipu.py

```diff
@@ -260,6 +263,16 @@ def to_tensor(x):
         args = apply_to_collection(args, dtype=(int, float), function=to_tensor)
         return args
 
+    def _disable_zero_grad(self) -> None:
+        lightning_module = self.lightning_module
+        if is_overridden("optimizer_zero_grad", lightning_module):
+            assert lightning_module is not None  # `is_overridden` returns False otherwise
+            rank_zero_warn(
+                "You have overridden the `LightningModule.optimizer_zero_grad` hook but it will be ignored since"
+                " IPUs handle the zeroing of gradients internally."
+            )
+            lightning_module.optimizer_zero_grad = None  # type: ignore[assignment]
```

Comment:
This should be restored at the end of execution to avoid patching tbh module with visible side effects to users

Reply:
Would this happen in the `teardown` function? Similar to how the PopTorch `DataLoader` is handled?

Reply:
Yes

Reply:
Ok, I'll implement that. 

@carmocca, `lightning_module.optimizer_zero_grad = None` can't be inside the `if is_overriden` block like it is in the commit. Otherwise the only way the IPU will be executed correctly is if the user overrides `optimizer_zero_grad` themselves.